### PR TITLE
Fixed #29188 -- Fixed size changing in ContentFile after string writing.

### PR DESCRIPTION
**Pull request by web-maker**
https://code.djangoproject.com/ticket/29188

### CODE REVIEW

File: django/core/files/base.py [hunk 1]
```diff
@@ -149,6 +149,12 @@ def open(self, mode=None):
     def close(self):
         pass
 
+    def write(self, data):
+        """ Write data and get size from underlying file """
+        result = self.file.write(data)
+        self._size = self._get_size_from_underlying_file()
```
Comment (charettes) [inline django/core/files/base.py]: I think invalidating the `size` cache would make more sense than re-computing the size on every single write.

```python
self.__dict__.pop('_size', None)
```

File: tests/files/tests.py [hunk 2]
```diff
@@ -206,6 +206,15 @@ def test_open_resets_file_to_start_and_returns_context_manager(self):
         with file.open() as f:
             self.assertEqual(f.read(), b'content')
 
+    def test_size_changing_after_writing(self):
+        """
+        ContentFile must change size after string writing.
+        """
+        f = ContentFile('')
+        self.assertFalse(f.size)
```
Comment (charettes) [inline tests/files/tests.py]: `assertEqual(f.size, 0)`.

File: tests/files/tests.py [hunk 3]
```diff
@@ -206,6 +206,15 @@ def test_open_resets_file_to_start_and_returns_context_manager(self):
...
+        f.write('Test string')
+        self.assertTrue(f.size > 0)
```
Comment (charettes) [inline tests/files/tests.py]: `assertEqual(f.size, 11)`.

File: django/core/files/base.py (hunk 1 above)
Comment (web-maker) [reply django/core/files/base.py]: @charettes yes, you are right. Invalidating is more better. Thanks for quick review!

**Pull request merged**